exp/api/v1alpha1/azureasomanagedcluster_types.go

```diff
@@ -31,6 +32,11 @@ const (
 // AzureASOManagedClusterSpec defines the desired state of AzureASOManagedCluster.
 type AzureASOManagedClusterSpec struct {
 	AzureASOManagedClusterTemplateResourceSpec `json:",inline"`
+
+	// ControlPlaneEndpoint is the location of the API server within the control plane. It fulfills Cluster
+	// API's cluster infrastructure provider contract.
```

Comment:
nit: Would it be more clear to clarify that it's not recommended for the user to set this? Similar to the comment here: https://github.com/willie-yao/cluster-api-provider-azure/blob/main/api/v1beta1/azurecluster_types.go#L48